src/db_options.rs

```diff
@@ -57,6 +57,13 @@ impl Cache {
         Cache(Arc::new(CacheWrapper { inner }))
     }
 
+    /// Creates an LRU cache with custom options.
+    pub fn new_lru_cache_opts(opts: &LruCacheOptions) -> Cache {
+        let inner =
+            NonNull::new(unsafe { ffi::rocksdb_cache_create_lru_opts(opts.inner) }).unwrap();
```

Comment:
Am I missing something or is there an issue in the code below?
```rust
let options = LruCacheOptions::default();
let cache = Cache::new_lru_cache_opts(&options);
drop(options);
// Now the cache holds a pointer to the freed memory.
```

Reply:
The pointer to the options is not stored in the resulting cache object; it is only borrowed for the duration of construction. See the following call chain:

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/db/c.cc#L4825

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/include/rocksdb/cache.h#L286

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/cache/lru_cache.cc#L689

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/cache/lru_cache.cc#L651C43-L651C47

Reply:
Thanks for the clarification!